main.js

```diff
@@ -395,53 +395,121 @@ module.exports = min
 
 
 //lesson-7
-const obj1 = {
-    a: 'a',
-    b: {
-        a: 'a',
-        b: 'b',
-        c: {
-            a: 1,
-        },
-    },
-};
-const obj2 = {
-    b: {
-        c: {
-            a: 1,
-        },
-        b: 'b',
-        a: 'a',
-    },
-    a: 'a',
-};
-const obj3 = {
-    a: {
-        c: {
-            a: 'a',
-        },
-        b: 'b',
-        a: 'a',
-    },
-    b: 'b',
-};
+// const obj1 = {
+//     a: 'a',
+//     b: {
+//         a: 'a',
+//         b: 'b',
+//         c: {
+//             a: 1,
+//         },
+//     },
+// };
+// const obj2 = {
+//     b: {
+//         c: {
+//             a: 1,
+//         },
+//         b: 'b',
+//         a: 'a',
+//     },
+//     a: 'a',
+// };
+// const obj3 = {
+//     a: {
+//         c: {
+//             a: 'a',
+//         },
+//         b: 'b',
+//         a: 'a',
+//     },
+//     b: 'b',
+// };
+//
+// function isEqual(object1, object2) {
+//         const props1 = Object.getOwnPropertyNames(object1);
+//         const props2 = Object.getOwnPropertyNames(object2);
+//         if (props1.length !== props2.length) {
+//             return false;
+//         }
+//         for (let i = 0; i < props1.length; i += 1) {
+//             const prop = props1[i];
+//             const bothAreObjects = typeof(object1[prop]) === 'object' && typeof(object2[prop]) === 'object';
+//             if ((!bothAreObjects && (object1[prop] !== object2[prop]))
+//                 || (bothAreObjects && !isEqual(object1[prop], object2[prop]))) {
+//                 return false;
+//             }
+//         }
+//
+//         return true;
+// }
+// console.log(isEqual(obj1, obj2))
+// console.log(isEqual(obj1, obj3))
 
-function isEqual(object1, object2) {
-        const props1 = Object.getOwnPropertyNames(object1);
-        const props2 = Object.getOwnPropertyNames(object2);
-        if (props1.length !== props2.length) {
-            return false;
-        }
-        for (let i = 0; i < props1.length; i += 1) {
-            const prop = props1[i];
-            const bothAreObjects = typeof(object1[prop]) === 'object' && typeof(object2[prop]) === 'object';
-            if ((!bothAreObjects && (object1[prop] !== object2[prop]))
-                || (bothAreObjects && !isEqual(object1[prop], object2[prop]))) {
-                return false;
-            }
-        }
 
-        return true;
+//lesson-8
+class User {
+    constructor(firstName, lastName) {
+    this.firstName = firstName;
+        this.lastName = lastName;
+    }
+    get fullName () {
+        return `${this.firstName} ${this.lastName}`;
+    }
+}
+
+class Student extends User {
+    constructor({firstName, lastName, admissionYear, courseName}) {
+        super(firstName, lastName);
+        this.admissionYear = admissionYear;
+        this.courseName = courseName;
+    }
+    get course () {
+        const currentYear = new Date().getFullYear();
+        return `${currentYear - this.admissionYear} курс`;
```

Comment:
I didn't run this code but I think maybe you can get 0th course when student started to study in 2022 (first course)